### ISSUE TITLE
[Android] "IllegalStateException: Design assumption violated" when adding/removing TabbedPage tabs at runtime

### ISSUE THREAD

**Issue by FlavioGoncalves-Cayas** [dotnet/maui#20802]
### Description

When adding/removing tabs to/from a TabbedPage a Java.Lang.IllegalStateException is thrown.
This only happens with Xamarin.AndroidX.RecyclerView >= 1.3.1. When you manage to pin Xamarin.AndroidX.RecyclerView package to <1.3.1 this exception is not thrown.

I have seen some discussion about this issue, namely in https://github.com/dotnet/maui/pull/17793 but that is closed and rather hard to find.

If you decide to close this issue feel free. Then at least people can find this if they stumble upon this issue.

### Steps to Reproduce

1. Clone the reproduction repository
2. Run it on Android
3. Wait 10 seconds for timer to remove the tab
4. Observe the crash mentioned above
5. Change Xamarin.AndroidX.RecyclerView version to 1.3.0.3
6. Run it on Android
7. Observe tab gets removed and added in 10 seconds interval without crash

### Link to public reproduction project repository

https://github.com/FlavioGoncalves-Cayas/AddRemoveTabDesignViolationRepro

### Version with bug

8.0.7 SR2

### Is this a regression from previous behavior?

Not sure, did not test other versions

### Last version that worked well

Unknown/Other

### Affected platforms

Android

### Affected platform versions

Android 14

### Did you find any workaround?

Use Xamarin.AndroidX.RecyclerView <1.3.1

**Comment by moljac** [comment]
I was able to repro with the repro sample from the repo.

**Comment by moljac** [comment]
@FlavioGoncalves-Cayas 

Is this look/design OK for MacCatalyst? Just asking cos images are bigger than tabs.

<img width="1025" alt="Screenshot 2024-02-25 at 15 50 30" src="https://github.com/dotnet/maui/assets/1768576/9b5fb35e-cc03-4e43-a9fc-d1dfb382d934">

**Comment by FlavioGoncalves-Cayas** [comment]
@moljac 
Personally for me I would say that's another bug, as I would expect the images to adjust their size automatically to the available space in the bottom tab bar. However this does not have to do with this issue and should probably be opened in another issue.

**Comment by moljac** [comment]
something is not right

<img width="451" alt="Screenshot 2024-02-25 at 15 54 02" src="https://github.com/dotnet/maui/assets/1768576/6987e867-d704-4656-818b-483d6003746a">

**Comment by moljac** [comment]
> @moljac Personally for me I would say that's another bug, as I would expect the images to adjust their size automatically to the available space in the bottom tab bar. However this does not have to do with this issue and should probably be opened in another issue.

Please do open new issue with affected platforms/idioms and the same sample.

**Comment by FlavioGoncalves-Cayas** [comment]
@moljac Looks like there already exists an issue for that https://github.com/dotnet/maui/issues/11984

**Comment by kevinxufei** [comment]
Verified this issue with Visual Studio 17.10.0 Preview 2. Can repro on Android platform with sample project.
![a](https://github.com/dotnet/maui/assets/86938401/5117e0df-cd84-4e68-afa0-1e5dbb8dfe9f)

**Comment by PureWeen** [comment]
Hopefully this will be resolved by
https://github.com/dotnet/maui/pull/20783

**Comment by maonaoda** [comment]
@PureWeen 
Hi, I had comfirmed that using [Xamarin.AndroidX.ViewPager2/1.1.0](https://www.nuget.org/packages/Xamarin.AndroidX.ViewPager2/1.1.0) fixed by [Bind using AndroidX.ViewPager2 1.1.0](https://github.com/xamarin/AndroidX/issues/905)
could fix this bug.

**Comment by maonaoda** [comment]
I think that following the modification of https://github.com/dotnet/maui/pull/20783, it is necessary to continue upgrading to meet the requirements of using [Xamarin.AndroidX.ViewPager2 1.1.0](https://www.nuget.org/packages/Xamarin.AndroidX.ViewPager2/1.1.0)

**Comment by moljac** [comment]
> @PureWeen Hi, I had comfirmed that using [Xamarin.AndroidX.ViewPager2/1.1.0](https://www.nuget.org/packages/Xamarin.AndroidX.ViewPager2/1.1.0) fixed by [Bind using AndroidX.ViewPager2 1.1.0](https://github.com/xamarin/AndroidX/issues/905) could fix this bug.

Sorry, but I will have to disappoint you. 

Tests and investigations I did in 

https://github.com/dotnet/maui/pull/23271

showed that it solved some/few exceptions, but not all.

**Comment by maonaoda** [comment]
I used a locally built Xamarin.AndroidX package and made some modifications based on https://github.com/dotnet/maui/pull/20783, including package`s upgrading and some warnings for deprecated code. I only confirmed that the "IllegalStateException: Design assumption violated" error did not occur again, but I did not execute all of maui's unit tests.

**Comment by moljac** [comment]
> I used a locally built Xamarin.AndroidX package and made some modifications based on #20783, including package`s upgrading and some warnings for deprecated code. I only confirmed that the "IllegalStateException: Design assumption violated" error did not occur again, but I did not execute all of maui's unit tests.

Yes. I reduced few errors with some fixes, but not all. I hoped update will solve everything, but no.

**Comment by HarishKumarSF4517** [comment]
This issue has been verified in Visual Studio Code 1.103.2 with MAUI latest version 9.0.110. It can be reproduced on the Android platform. I have attached the snap of exception thrown:

![Image](https://github.com/user-attachments/assets/3e367c03-8473-426d-be9c-eb40be71257e)
